src/Application.cpp

```diff
@@ -0,0 +1,743 @@
+                        std::string locationName = nodeIdName.substr(13, posDot - 13);
+                        // Search for the location in the list of locations inside the country of the web service.
+                        auto searchLocation = LocationData{ locationName, countryCode };
+                        auto itLocation = std::find(itCountry->getLocations().begin(), itCountry->getLocations().end(), searchLocation);
+
+                        /* While the location is not found, continue checking for dots within its name. */
+                        while (itLocation == itCountry->getLocations().end()) {
+                            posDot = nodeIdName.find(".", posDot + 1);
+                            locationName = nodeIdName.substr(13, posDot - 13);
+                            // Search for the location in the list of locations inside the country of the web service.
+                            searchLocation = LocationData{ locationName, countryCode };
+                            itLocation = std::find(itCountry->getLocations().begin(), itCountry->getLocations().end(), searchLocation);
+                        }
+
+                        std::string locationObjNameId = countryObjNameId + "." + locationName;
+                        /*
+                        The location name was found correctly. Check if there are more characters after the location name comparing the full node id name's size to the node id until the location name.
+                        */
+                        if (nodeIdName.size() > locationObjNameId.size()) {
+                            UA_NodeId locationObjId = UA_NODEID_STRING_ALLOC(WebService::OPC_NS_INDEX, locationObjNameId.c_str());
+                            /*
+                            Only try to download weather data if they not exist in the address space.
+                            The isAddingWeatherToAddressSpace boolean variable in LocationData controls when we are adding the weather data in the OPC UA address space, that being said the requestWeather function bellow will not be called more than once.
+                            */
+                            if (itLocation->getIsInitialized() && !(itLocation->getHasBeenReceivedWeatherData()) && !(itLocation->getIsAddingWeatherToAddressSpace())) {
+                                requestWeather(webService->getServer(), *itLocation, locationObjId);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        return defaultGetNode(nodestoreContext, nodeId);
+    }
+}
+
+int main(int argc, char* argv[]) {
+
+    //extracting current directory
+    std::string fullPath(argv[0]);
+    std::string currentDir = fullPath.substr(0, fullPath.find_last_of("/\\") + 1); //including separator since we are adding file name later
+    std::cout << "Current directory: " << currentDir << std::endl;
```

Comment:
Current directory is not correct naming here. This is _executable directory_.
It is possible to run the program by full path from some other directory and that other directory will considered as current one.